content/OTP/Specifications/OTP_validation_protocol.adoc

```diff
@@ -115,8 +119,45 @@ counters:
  https://api.yubico.com/wsapi/2.0/verify?id=87&otp=vvvvvvcucrlcietctckflvnncdgckubflugerlnr&timeout=8&sl=50&nonce=askjdnkajsndjkasndkjsnad&timestamp=1
 
 
+If you are using the public YubiCloud service (api.yubico.com), you
+MUST use HTTP over TLS and you MUST use Server Name Indication. You
+SHOULD NOT hard-code the IP addresses that the DNS name resolves to,
+and you SHOULD NOT pin the TLS certificates that the service presents.
+You SHOULD sign your API requests and validate the response signatures
+as described above.
+
 === Response
 
+If the verification server has successfully processed your request
+(even if the reponse is not a successful verification), it will return
+an HTTP status of `200 OK` with a text/plain body.
+
+If you get a `4xx` or `5xx` response you should retry your request a
```

Comment:
Should `4xx` be retried without modifying the request?

Reply:
Yes - we've seen an instance of the AWS API Gateway returning a 400 when its request-rewriting engine had an internal error (instead of the 500 we would expect).

Since all valid responses in the V2 API will be a 200 status code even if they're an error at the application level, a 400 response should always be retried.

When we do the V3 API we will not be using the API Gateway's request-rewriting engine, so this will not be an issue.

Reply:
:+1: